src/popper/runner_kubernetes.py

```diff
@@ -110,10 +110,19 @@ def _init_pod_schedule(self):
             pod_host_node = self._config.resman_opts.pod_host_node
 
         elif not self._config.resman_opts.get("persistent_volume_name", None):
-            nodes = [
-                node.metadata.labels["kubernetes.io/hostname"]
-                for node in self._kclient.list_node().items
-            ]
+            nodes = []
+
+            for node in self._kclient.list_node().items:
+                node_role = ""
+                if node.spec.taints is not None:
+                    node_role = node.specs.taints.key
+                if node_role != "node-role.kubernetes.io/master":
```

Comment:
```
if node_role != "node-role.kubernetes.io/master" and node_role != "node.kubernetes.io/unreachable":
```


We can also check for the `node.kubernetes.io/unreachable` label to check for unreachable nodes.

Reply:
Should unreachable nodes be added in the else case below this code or should they be excluded there as well?

Reply:
the unreachable label can be excluded in line 119 as well.